Sources/StreamChatUI/ChatMessageList/RETHINK/MessageContentView.swift

```diff
@@ -0,0 +1,543 @@
+    /// otherwise it will be added to `bubbleThreadMetaContainer`.
+    public private(set) lazy var bubbleContentContainer = ContainerStackView(axis: .vertical)
+        .withoutAutoresizingMaskConstraints
+
+    /// The container which holds `threadArrowView`, `threadAvatarView`, and `threadReplyCountButton`
+    public private(set) var threadInfoContainer: ContainerStackView?
+
+    /// Makes sure the `layout(options: MessageLayoutOptions)` is called just once.
+    /// - Parameter options: The options describing the layout of the content view.
+    open func setUpLayoutIfNeeded(options: MessageLayoutOptions) {
+        guard layoutOptions == nil else {
+            log.assert(layoutOptions == options, """
+            Attempt to setup "\(options)" layout for \(self) while it has already been laid out with "\(layoutOptions!)" options.
+            `MessageContentView` is supposed to be laid out only once.
+            """)
+            return
+        }
+        layout(options: options)
+        layoutOptions = options
+    }
+
+    /// Instantiates the subviews and laid them out based on the received options.
+    /// - Parameter options: The options describing the layout of the content view.
+    open func layout(options: MessageLayoutOptions) {
+        var constraintsToActivate: [NSLayoutConstraint] = []
+
+        // Main container
+        mainContainer.alignment = .trailing
+        mainContainer.isLayoutMarginsRelativeArrangement = true
+        mainContainer.layoutMargins.top = 0
+
+        addSubview(mainContainer)
+        constraintsToActivate += [
+            mainContainer.bottomAnchor.pin(equalTo: bottomAnchor),
+            mainContainer.widthAnchor.pin(
+                lessThanOrEqualTo: widthAnchor,
+                multiplier: maxContentWidthMultiplier
+            )
+        ]
+        
+        if options.contains(.flipped) {
+            constraintsToActivate += [
+                mainContainer.trailingAnchor
+                    .pin(equalTo: trailingAnchor)
+                    .almostRequired
+            ]
+        } else {
+            constraintsToActivate += [
+                mainContainer.leadingAnchor
+                    .pin(equalTo: leadingAnchor)
+                    .almostRequired
+            ]
+        }
+
+        // Avatar view
+        if options.contains(.avatar) {
+            let avatarView = createAvatarView()
+            constraintsToActivate += [
+                avatarView.widthAnchor.pin(equalToConstant: 32),
+                avatarView.heightAnchor.pin(equalToConstant: 32)
+            ]
+
+            mainContainer.addArrangedSubview(avatarView)
+        }
+        
+        if options.contains(.avatarSizePadding) {
+            let spacer = UIView().withoutAutoresizingMaskConstraints
+            spacer.isHidden = true
+            constraintsToActivate += [spacer.widthAnchor.pin(equalToConstant: 32)]
+            mainContainer.addArrangedSubview(spacer)
+        }
```

Comment:
Since the avatar size is used in two places, it would be nice to extract it into a variable

Reply:
✅ Done!